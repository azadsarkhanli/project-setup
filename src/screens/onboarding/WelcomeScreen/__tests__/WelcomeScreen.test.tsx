import renderer from '@arkapp/test-utils/test-renderer';
import React from 'react';

import { WelcomeScreen } from '..';

function onPress() {
  console.log('onPress');
}

test('renders Welcome screen correctly', () => {
  const tree = renderer
    .create(<WelcomeScreen onLogin={onPress} onRegister={onPress} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
