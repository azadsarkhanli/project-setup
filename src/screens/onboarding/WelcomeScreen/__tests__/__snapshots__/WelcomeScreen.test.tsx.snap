// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Welcome screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#001f18",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 1,
          "flexShrink": 0,
        }
      }
      keyboardShouldPersistTaps="handled"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
                "marginBottom": 2,
                "marginLeft": 6,
                "marginRight": 6,
                "marginTop": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "paddingLeft": 56,
                  "paddingRight": 56,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": "auto",
                    "marginRight": "auto",
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={36}
                  bbWidth={36}
                  fill="none"
                  focusable={false}
                  height={36}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "flex": 0,
                        "height": 36,
                        "width": 36,
                      },
                    ]
                  }
                  vbHeight={36}
                  vbWidth={36}
                  width={36}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup
                      clipPath="a"
                    >
                      <RNSVGPath
                        d="M35.966 17.987a17.983 17.983 0 1 1-35.965 0 17.983 17.983 0 0 1 35.965 0Z"
                        fill={4294967295}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      />
                      <RNSVGPath
                        d="M17.983 0C8.05 0 0 8.056 0 17.995 0 27.935 8.05 36 17.983 36c7.389 0 13.734-4.465 16.5-10.843h-.798v-4.333h.742v3.68h.326a17.923 17.923 0 0 0 1.214-6.51c0-5.36-2.343-10.167-6.058-13.462l-9.981 1.835V4.51l2.462-.332v-2.1l-2.462-.343V.1C19.282.032 18.633 0 17.983 0Zm6.08 2.354v1.548l4.205-.675c-.137-.093-.277-.185-.416-.276l-3.788-.597Zm-4.146 4.753h11.15V9.76c0 2.146-.91 3.272-3.283 3.272-1.79 0-2.528-.722-2.82-1.37l-5.047 1.635V11.35l4.63-1.359a9.425 9.425 0 0 1-.045-.984h-4.585v-1.9Zm6.137 2.012v.64c0 1.036.404 1.449 1.73 1.449s1.698-.413 1.698-1.448v-.64h-3.428v-.001Zm-6.137 5.195h11.15v1.934H26.94v.044l4.125 2.012v1.89l-4.17-2.078-6.98 2.343v-1.934l5.204-1.547v-.045l-1.236-.685h-3.967v-1.934Zm-4.328 1.746c.78 0 1.158.494 1.158 1.106v2.232c0 .613-.378 1.105-1.158 1.105-.778 0-1.112-.492-1.112-1.105v-2.232c0-.613.334-1.106 1.112-1.106Zm-3.934.078h1.113c.797 0 1.158.4 1.158 1.138 0 .372-.18.761-.54.907.38.167.619.47.619 1.027 0 .715-.314 1.25-1.158 1.25h-1.192v-4.323Zm5.632 0h.73v1.59l.82-1.59h.732l-.81 1.59.888 2.731h-.742l-.618-2.012-.27.465v1.547h-.73v-4.321Zm-1.799.574c-.217.037-.28.204-.28.487v2.2c0 .324.081.452.381.452s.427-.128.427-.453v-2.2c0-.322-.127-.485-.427-.485-.037 0-.07-.006-.1 0Zm-3.102.033v1.172h.27c.282 0 .506-.204.506-.564 0-.401-.133-.608-.461-.608h-.315Zm0 1.78v1.326h.315c.397 0 .54-.236.54-.674 0-.444-.143-.652-.585-.652h-.27Zm7.53 2.321h.765v1.747h.775v-1.747h.731v4.322h-.731v-1.934h-.775v1.934h-.765v-4.322Zm3.474 0h.921l.776 4.322h-.697l-.157-.984h-.81l-.112.984h-.73l.809-4.322Zm2.079 0h.697l.888 2.465v-2.465h.697v4.322h-.664l-.921-2.73h-.045v2.73h-.652v-4.322Zm2.934 0h1.123c.767 0 1.158.396 1.158 1.282v1.714c0 .888-.39 1.326-1.158 1.326h-1.123v-4.322Zm2.82 0h2.001v.64h-1.236v1.107h.967v.64h-.967v1.294h1.315v.64h-2.08v-4.32Zm-2.09.608v3.106h.27c.395 0 .54-.152.54-.608v-1.89c0-.455-.145-.608-.54-.608h-.27Zm-5.327.221-.27 1.868h.584l-.27-1.868h-.044Z"
                        fill={4293618176}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      />
                    </RNSVGGroup>
                    <RNSVGDefs>
                      <RNSVGClipPath
                        name="a"
                      >
                        <RNSVGPath
                          d="M0 0h36v36H0z"
                          fill={4294967295}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                        />
                      </RNSVGClipPath>
                    </RNSVGDefs>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              style={Object {}}
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "flex-end",
                "marginBottom": 24,
                "marginLeft": 24,
                "marginRight": 24,
                "marginTop": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={300}
                bbWidth={300}
                fill="none"
                focusable={false}
                height={300}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 300,
                      "width": 300,
                    },
                  ]
                }
                vbHeight={300}
                vbWidth={300}
                width={300}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M62.19 271.929c28.728-21.385 83.508-52.76 135.921-74.435.929-.894 1.873-1.794 2.833-2.703 29.678 0 50.147-45.88 53.6-73.724 3.401-27.905-18.414-58.248-45.102-74.636-26.689-16.33-58.247-18.772-89.866-18.84-31.618-.068-63.294 2.198-79.055 20.661-15.687 18.41-15.46 53.068-17.84 84.69-2.38 31.622-7.421 60.258.226 87.066 6.25 22.062 21.134 42.967 39.282 51.921Z"
                    fill={4294961622}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M217.839 189.759c-2.784-1.025-11.891-13.98-11.891-29.353 0-17.862 17.571-51.67 19.241-63.714 1.064-7.66 5.861-75.144 42.29-75.144 19.864 0 21.521 41.762 21.521 60.44 0 43.634-59.004 80.801-59.004 103.466a85.823 85.823 0 0 1-12.157 4.305Z"
                    fill={4294961622}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M269.795 65.037a2.655 2.655 0 0 1 .092 3.754c-10.306 10.824-21.566 31.964-30.273 54.989-8.695 22.996-14.641 47.332-14.641 64.359a2.655 2.655 0 1 1-5.31 0c0-17.965 6.197-43.001 14.984-66.237 8.776-23.209 20.339-45.162 31.394-56.773a2.655 2.655 0 0 1 3.754-.093ZM7.673 155.152a2.656 2.656 0 0 1 3.607-1.048c5.418 2.98 12.475 10.122 19.849 18.994 7.465 8.982 15.526 20.075 22.952 31.331 7.426 11.255 14.251 22.723 19.232 32.465 2.489 4.868 4.537 9.344 5.97 13.172 1.403 3.747 2.323 7.145 2.323 9.748a2.655 2.655 0 1 1-5.31 0c0-1.593-.62-4.235-1.987-7.886-1.337-3.571-3.29-7.852-5.725-12.617-4.87-9.523-11.587-20.819-18.936-31.957-7.348-11.138-15.293-22.065-22.604-30.861-7.401-8.905-13.89-15.297-18.324-17.735a2.655 2.655 0 0 1-1.047-3.606Z"
                    fill={4294889124}
                    fillRule={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M86.526 205.724a2.656 2.656 0 0 1 1.257 3.539c-1.277 2.686-2.206 6.764-2.819 11.654-.605 4.832-.878 10.231-.954 15.419-.153 10.405.484 19.723.71 21.558a2.655 2.655 0 1 1-5.27.649c-.265-2.143-.906-11.716-.75-22.285.078-5.299.356-10.908.995-16.002.631-5.037 1.64-9.802 3.292-13.274a2.656 2.656 0 0 1 3.539-1.258Z"
                    fill={4294889124}
                    fillRule={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M272.626 174.81a1.992 1.992 0 0 1-1.606 2.314c-39.229 7.074-91.69 29.818-136.284 53.705-22.271 11.929-42.519 24.111-58.1 34.707-7.793 5.299-14.397 10.187-19.496 14.436-5.142 4.286-8.613 7.809-10.282 10.372a1.992 1.992 0 0 1-3.338-2.174c2.015-3.094 5.883-6.934 11.07-11.257 5.23-4.36 11.947-9.327 19.806-14.671 15.719-10.689 36.09-22.942 58.46-34.925 44.688-23.937 97.597-46.925 137.457-54.113a1.991 1.991 0 0 1 2.313 1.606Z"
                    fill={4294889124}
                    fillRule={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M121.091 56.6c0-2.375 1.791-4.301 4-4.301h31.978c2.21 0 4.001 1.926 4.001 4.301v71.324c0 2.375-1.791 4.301-4.001 4.301h-31.978c-2.209 0-4-1.926-4-4.301V56.6Z"
                    fill={4290924032}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="m115.574 261.083-14.581 7.202v3.943h23.535v-9.32l-8.954-1.825ZM176.058 255.148a89.554 89.554 0 0 0-.478-1.177l-6.264 2.849-13.367 11.001 1.247 4.387h10.64l11.744-7.939s-2.166-5.729-3.5-9.067l-.022-.054Z"
                    fill={4280881152}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M134.9 50.514c5.481 0 9.924-6.025 9.924-13.456 0-7.432-4.443-13.456-9.924-13.456s-9.924 6.024-9.924 13.456c0 7.431 4.443 13.456 9.924 13.456ZM127.675 13.024c-1.3-1.921-4.97-1.487-8.197 1.011a11.811 11.811 0 0 0-3.053 3.466.848.848 0 0 0-.334-.082c-1.567 0-27.611 14.98-27.365 21.89.247 6.91 8.984 17.994 12.034 20.12 0 0 13.764-6.911 13.024-9.13-.74-2.219-8.667-9.32-10.057-10.725 1.634-.917 14.708-14.847 16.184-16.424a10.719 10.719 0 0 0 4.284-2.082c3.22-2.502 4.784-6.101 3.48-8.044Z"
                    fill={4294751626}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M120.192 71.554c3-2.595-1.967-24.356-7.29-24.89-5.324-.534-14.608 10.251-14.235 13.312.374 3.061 13.945 18.102 21.525 11.578Z"
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M108.282 123.655c0 3.585 1.666 127.607 4.667 138.36 2.666 1.434 9.667 1.792 12 .717 2.334-1.076 17.225-137.113 17.225-143.084 0-5.972-33.892 4.007-33.892 4.007Z"
                    fill={4290924032}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M168.066 258.139c-3.183-5.377-44.636-109.784-47.886-135.528 3.403-2.835 28.732-7.355 30.782-2.208 2.05 5.147 26.668 132.406 25.825 134.058-.844 1.653-6.327 4.23-8.721 3.678Z"
                    fill={4293618176}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M135.04 54.475c21.501 0 20.168 63.982 15.501 67.925-4.667 3.943-41.17 4.122-43.503 1.254-2.334-2.867-7.847-69.18 28.002-69.18Z"
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M150.685 32.488c0-2.727-2.423-4.695-2.423-6.208s.83-4.223-1.027-6.162c-2.257-2.355-5.854-1.372-9.074-1.372-3.027 0-4.697-3.782-8.827-3.782-2.753 0-2.403 3.842-3.433 4.796-.764.717-6.714 2.093-6.714 5.018 0 3.27.08 7.657 6.393 7.657.421-1.57 2.001-5.019 3.801-6.553.926.43 6 2.681 7.257 3.08 1.257.397 3.093 6.182 5.04 6.881 1.947.7 2.07 4.552 3.494 4.388 1.423-.165 5.513-.81 5.513-7.743Z"
                    fill={4293618176}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M143.445 40.94c1.448 0 2.623-1.263 2.623-2.82 0-1.559-1.175-2.822-2.623-2.822-1.449 0-2.624 1.263-2.624 2.821 0 1.558 1.175 2.821 2.624 2.821ZM139.861 59.492c-.42 1.56-9.237-.334-9.067-2.097l1.507-15.664 10.624 2.459s-2.637 13.75-3.064 15.302Z"
                    fill={4294751626}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M200.775 63.328a4.646 4.646 0 0 1 3.71 5.424L167.2 267.5a4.647 4.647 0 1 1-9.135-1.714l37.286-198.746a4.646 4.646 0 0 1 5.424-3.71Z"
                    fill={4283572480}
                    fillRule={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M193.938 122.738c-.993-1.08-2.275-1.801-3.667-2.061-4.447-7.718-19.558-31.504-26.221-40.146-4.804 3.71-10.668 10.197-10.668 10.197 2.45 2.954 20.802 26.3 30.565 35.454-.005.959.169 1.91.513 2.795a7.158 7.158 0 0 0 1.488 2.356c2.47 2.66 6.263 2.889 8.47.52 2.207-2.369 1.99-6.456-.48-9.115Z"
                    fill={4294751626}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M143.483 64.96c8.624-6.894 22.842 9.774 22.608 14.158-.233 4.383-11.16 14.105-14.897 14.115-4.564.007-15.051-22.406-7.711-28.274Z"
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#fafaf9",
                    "fontFamily": "HaasGrotTextRApp-65Medium",
                    "fontSize": 24,
                    "fontStyle": undefined,
                    "fontWeight": undefined,
                    "letterSpacing": 0,
                    "lineHeight": 30,
                    "marginBottom": 16,
                    "marginTop": 48,
                    "textAlign": "center",
                    "textDecorationLine": undefined,
                  }
                }
              >
                Les og lytt til dine favorittbøker
              </Text>
              <Text
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#f0f4f3",
                    "fontFamily": "HaasGrotTextRApp-55Roman",
                    "fontSize": 16,
                    "fontStyle": undefined,
                    "fontWeight": undefined,
                    "letterSpacing": 0,
                    "lineHeight": 24,
                    "marginBottom": 24,
                    "textAlign": "center",
                    "textDecorationLine": undefined,
                  }
                }
              >
                E-bøkene og lydbøkene du har kjøpt på ARK.no er automatisk tilgjengelig i appen
              </Text>
            </View>
            <View
              style={Object {}}
            />
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                _disabled={
                  Object {
                    "_text": Object {
                      "color": "darkText",
                    },
                    "bg": false,
                    "opacity": "50",
                  }
                }
                _loading={
                  Object {
                    "bg": "primary.300",
                    "opacity": "50",
                  }
                }
                _text={
                  Object {
                    "color": "lightText",
                  }
                }
                accessibilityLabel="\\"Logg inn\\" knapp"
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onHoverIn={[Function]}
                onHoverOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#c24e00",
                    "borderRadius": 9999,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 12,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 12,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      style={
                        Object {
                          "backgroundColor": undefined,
                          "color": "#f3f7f6",
                          "fontFamily": "HaasGrotTextRApp-65Medium",
                          "fontSize": 16,
                          "fontStyle": undefined,
                          "fontWeight": undefined,
                          "letterSpacing": 0,
                          "lineHeight": 24,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Logg inn
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": 20,
                  }
                }
              />
              <View
                _disabled={
                  Object {
                    "opacity": "50",
                  }
                }
                _loading={
                  Object {
                    "opacity": "80",
                  }
                }
                _text={
                  Object {
                    "color": "darkText",
                  }
                }
                accessibilityLabel="\\"Opprett bruker\\" knapp"
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onHoverIn={[Function]}
                onHoverOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ffe9d6",
                    "borderRadius": 9999,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 12,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 12,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      style={
                        Object {
                          "backgroundColor": undefined,
                          "color": "#fef7f1",
                          "fontFamily": "HaasGrotTextRApp-65Medium",
                          "fontSize": 16,
                          "fontStyle": undefined,
                          "fontWeight": undefined,
                          "letterSpacing": 0,
                          "lineHeight": 24,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Opprett bruker
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
